sct-commons/src/main/java/org/lfenergy/compas/sct/commons/util/FcdaCandidatesHelper.java

```diff
@@ -0,0 +1,105 @@
+// SPDX-FileCopyrightText: 2022 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.util;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import org.apache.commons.lang3.StringUtils;
+import org.lfenergy.compas.sct.commons.dto.ResumedDataTemplate;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public enum FcdaCandidatesHelper implements Predicate<ResumedDataTemplate> {
+    SINGLETON;
+
+    private static final String FCDA_CONSTRAINTS_FILE_NAME = "FcdaCandidates.csv";
+    public static final int NUMBER_OF_COLUMNS = 4;
+    private static final int COLUMN_LN_CLASS = 0;
+    private static final int COLUMN_DO_NAME = 1;
+    private static final int COLUMN_DA_NAME = 2;
+    private static final int COLUMN_FC = 3;
+    private static final String DELIMITER = ";";
+    private static final String COMMENT = "#";
+
+    private Set<FcdaCandidate> fcdaCandidates;
+
+    /**
+     * Alias of isFcdaCandidate(ResumedDataTemplate) to implement Predicate
+     * @see #isFcdaCandidate(ResumedDataTemplate)
+     */
+    @Override
+    public boolean test(ResumedDataTemplate resumedDataTemplate) {
+        return isFcdaCandidate(resumedDataTemplate);
+    }
+
+    /**
+     * Check if resumedDataTemplate is isFcdaCandidate
+     * @see #isFcdaCandidate(ResumedDataTemplate)
+     */
+    public boolean isFcdaCandidate(ResumedDataTemplate resumedDataTemplate) {
+        return isFcdaCandidate(resumedDataTemplate.getLnClass(),
+            resumedDataTemplate.getDoName().toStringWithoutInst(),
+            resumedDataTemplate.getDaName().toString(),
+            resumedDataTemplate.getFc().value());
+    }
+
+    private boolean isFcdaCandidate(String lnClass, String doName, String daName, String fc) {
+        if (StringUtils.isBlank(lnClass)
+            || StringUtils.isBlank(doName)
+            || StringUtils.isBlank(daName)
+            || StringUtils.isBlank(fc)) {
+            throw new IllegalArgumentException("parameters must not be blank");
+        }
+
+        if (fcdaCandidates == null) {
+            fcdaCandidates = readFcdaCandidatesFile();
+        }
+        return fcdaCandidates.contains(new FcdaCandidate(lnClass, doName, daName, fc));
+    }
+
+    private Set<FcdaCandidate> readFcdaCandidatesFile() {
```

Comment:
I know you think it's a simple case, but here you could use the `openCsv` librairy to more readability;  
it's very simple : 
```xml
<!-- opencsv utilisé pour la lecture des données mockés de referentiels -->
		<dependency>
			<groupId>com.opencsv</groupId>
			<artifactId>opencsv</artifactId>
			<version>5.5.2</version>
		</dependency>
```
and you method become : 
```java
private List<FcdaCandidate2> readFcdaCandidatesFile() {
        InputStreamReader inputStreamReader = new InputStreamReader(getClass().getClassLoader().getResourceAsStream(FCDA_CONSTRAINTS_FILE_NAME), StandardCharsets.UTF_8);
        ColumnPositionMappingStrategy<FcdaCandidate2> ms = new ColumnPositionMappingStrategy<>();
        ms.setType(FcdaCandidate2.class);
        return new CsvToBeanBuilder<FcdaCandidate2>(inputStreamReader)
                .withSeparator(';')
                .withSkipLines(5)
                .withType(FcdaCandidate2.class)
                .withFieldAsNull(CSVReaderNullFieldIndicator.EMPTY_SEPARATORS)
                .withMappingStrategy(ms)
                .build()
                .parse();
    }
```
with 
```java
@EqualsAndHashCode
@Getter
@Setter
public class FcdaCandidate2 {
    @CsvBindByPosition(position = 0)
    private String lnClass;
    @CsvBindByPosition(position = 1)
    private String doName;
    @CsvBindByPosition(position = 2)
    private String daName;
    @CsvBindByPosition(position = 3)
    private String fc;
}
```